include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -420,6 +421,9 @@ __get_accessor_ptr(const _Acc& __acc)
 #endif
 }
 
+template <typename... _Ts>
+struct __is_device_copyable: std::conjunction<sycl::is_device_copyable<_Ts>...> {};
```

Comment:
I don't feel this belong to this file and to current namespace. It seems like the better place is `sycl_traits.h`. The reason for that is here we check the some compiler/ SYCL_LIBRARY versions, while this trait exist unconditionally. It's not the super strong preference, though, so if you have counter arguments I am happy to hear them.

Reply:
I agree.  I'm not sure why it was put there in the first place.  I don't think we will have use for it outside `sycl_traits.h`. I put it into the `sycl_traits.h`, into a unnamed namespace.

Reply:
It was put there because I suggested it :) (As a general recommendation, I think it makes sense to skim through the comments of others, e.g. most recent ones, to see which questions have been discussed).

Essentially, the idea of `namespace __dpl_sycl` is to replace/complement/handle version differences for the SYCL routines, keeping their key semantics intact. This one is not much different really. But I am OK with the current code either.

Reply:
Ah, I missed that discussion above, apologies.  
I understand.  Yes, I suppose essentially we are enhancing the existing `sycl::is_device_copyable` trait with this struct by allowing it to accept a pack of types, rather than a single one at a time.  I was merely looking at this as a helper for our device_copyable specializations.  

Do you see us wanting to use `__is_device_copyable` outside of `sycl_traits.h`?  If so I think that is reason to move it back.  My understanding was that we wanted to keep all the device copyable specializations here in `sycl_traits.h`, so having the helper here doesn't encourage usage elsewhere.

Reply:
I don't see a use case where we would want this either (other than use within these specializations in `sycl_traits.h`).  My preference is to keep it as is.